### PR TITLE
NAS-117082 / 22.12 / NAS-117082: Improving loading indication on Datasets and Devices

### PR DESCRIPTION
**Pull request by undsoft**
Testing: go to Datasets and Devices.
If you go to Datasets, somewhere else and back, the tree will still be populated with old data. Same happens when you perform an action on a tree (like adding/removing a dataset).
This is a deliberate choice.

Also renames AppLoaderModule to LoaderModule.

### CODE REVIEW

File: src/app/pages/datasets/store/dataset-store.service.ts [hunk 1]
```diff
@@ -25,6 +27,7 @@ const initialState: DatasetTreeState = {
 @Injectable()
 export class DatasetTreeStore extends ComponentStore<DatasetTreeState> {
   readonly isLoading$ = this.select((state) => state.isLoading);
+  // TODO
```
Comment (RehanY147) [inline src/app/pages/datasets/store/dataset-store.service.ts:30]: What's TODO?

**Review by RehanY147 — CHANGES_REQUESTED** [review]
If you move around the dashboard a couple of times, the pools dashboard is loading but shows no indication. There's no errors in the console. But no loading indication either.
loading issue pools

File: src/app/pages/datasets/store/dataset-store.service.ts (hunk 1 above)
Comment (undsoft) [reply src/app/pages/datasets/store/dataset-store.service.ts:30]: This is a TODO for next line to show error somewhere. Not included in this PR.

**Comment by undsoft** [comment]
@RehanY147 Didn't work on loading indication for Storage Dashboard in this PR. Just Datasets and Devices (as in list of Devices).

**Review by dszidi — COMMENTED** [review]
When I go between Storage 2.0 link and the devices page (via topology card) I get a console error.

image

EDIT: Nvm, this also seems to be there in master branch.

**Pull request merged**